modules/openapi-generator/src/main/java/org/openapitools/codegen/utils/ModelUtils.java

```diff
@@ -2169,26 +2167,6 @@ public static boolean isParent(Schema schema) {
         return false;
     }
 
-    /**
-     * Returns a clone of the schema.
-     *
-     * @param schema the schema.
-     * @param specVersionGreaterThanOrEqualTo310 true if spec version is 3.1.0 or later.
-     * @return a clone of the schema.
-     */
-    public static Schema cloneSchema(Schema schema, boolean specVersionGreaterThanOrEqualTo310) {
-        Schema clone = AnnotationsUtils.clone(schema, specVersionGreaterThanOrEqualTo310);
-
-        // check to see if type is set and clone it if needed
-        // in openapi-generator, we also store type in `type` for 3.1 schema
-        // to make it backward compatible with the rest of the code base.
-        if (schema.getType() != null) {
-            clone.setType(schema.getType());
```

Comment:
looks like this logic is completely removed, right?

Reply:
SchemaUtils.cloneSchema must not lose a type info after clone (instead of AnnotationUtils), so yep, I've removed it :)

Reply:
👍 

FYI. we've refactored the code a bit and no longer using `type` to store the actualy type in 3.1 spec: https://github.com/OpenAPITools/openapi-generator/pull/18577